Tests/SecureXPCTests/Client & Server/XPCRequestContext Tests.swift

```diff
@@ -0,0 +1,131 @@
+            contextValueAccessed.fulfill()
+        }
+        server.start()
+        
+        client.send(route: route, onCompletion: nil)
+        
+        self.waitForExpectations(timeout: 1)
+    }
+    
+    func testGetEffectiveGroupId_async() async throws {
+        let route = XPCRoute.named("does", "nothing")
+        let server = XPCServer.makeAnonymous()
+        let client = XPCClient.forEndpoint(server.endpoint)
+        
+        server.registerRoute(route) { () async -> Void in
+            XCTAssertNotNil(XPCRequestContext.effectiveGroupId)
+        }
+        server.start()
+        
+        try await client.send(route: route)
+    }
+    
+    func testGetEffectiveGroupId_sync() throws {
+        let route = XPCRoute.named("does", "nothing")
+        let server = XPCServer.makeAnonymous()
+        let client = XPCClient.forEndpoint(server.endpoint)
+        
+        let contextValueAccessed = self.expectation(description: "Able to access effective group id")
+        
+        server.registerRoute(route) {
+            // If this fails it'll fatalError, so all we're doing here is ensuring that doesn't happen
+            _ = XPCRequestContext.effectiveGroupId
+            contextValueAccessed.fulfill()
+        }
+        server.start()
+        
+        client.send(route: route, onCompletion: nil)
+        
+        self.waitForExpectations(timeout: 1)
+    }
+    
+    func testGetClientCode_async() async throws {
+        let route = XPCRoute.named("does", "nothing")
+        let server = XPCServer.makeAnonymous()
+        let client = XPCClient.forEndpoint(server.endpoint)
+        
+        server.registerRoute(route) { () async -> Void in
+            XCTAssertNotNil(XPCRequestContext.clientCode)
+            var staticCode: SecStaticCode?
+            SecCodeCopyStaticCode(XPCRequestContext.clientCode!, [], &staticCode)
+            var signingInfo: CFDictionary?
+            SecCodeCopySigningInformation(staticCode!, [], &signingInfo)
+            if let signingInfo = signingInfo as NSDictionary?,
+               signingInfo[kSecCodeInfoIdentifier] as? String == "com.apple.xctest" {
+                // success
+                return
+            } else {
+                XCTFail("wrong identifier")
+            }
+        }
+        server.start()
+        
+        try await client.send(route: route)
+    }
+    
+    func testGetClientCode_sync() throws {
+        let route = XPCRoute.named("does", "nothing")
+        let server = XPCServer.makeAnonymous()
+        let client = XPCClient.forEndpoint(server.endpoint)
+        
+        let clientCodeNotNil = self.expectation(description: "Client code should not be nil in this circumstance")
+        let xctestIdentifier = self.expectation(description: "Client code's identifier should be com.apple.xctest")
+        
+        server.registerRoute(route) {
+            if let clientCode = XPCRequestContext.clientCode {
```

Comment:
Use [`XCTUnwrap(_:_:file:line:)`](https://developer.apple.com/documentation/xctest/3380195-xctunwrap):

```swift
let clientCode = XCTUnwrap(XPCRequestContext.clientCode)
...
```

Reply:
Ah cool, didn't know that existed

Reply:
Oh but it `throws`, and I need to use it inside of the handler's closure so it'd just get swallowed :(